authapi/tests/test_teams.py

```diff
@@ -695,6 +739,136 @@ def test_remove_permission_from_team(self):
+        to remove permissions from any teams.'''
+        _, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        org = SeedOrganization.objects.create(title='test org')
+        team = SeedTeam.objects.create(title='test team', organization=org)
+
+        data = {
+            'type': 'foo:bar',
+            'object_id': '2',
+            'namespace': 'foo',
+        }
+        permission = SeedPermission.objects.create(**data)
+        response = self.client.delete(reverse(
+            'seedteam-permissions-detail', args=[team.id, permission.id]))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_permission_remove_permission_read_access(self):
+        '''Any user that has read access to a team can remove team permissions
+        that aren't org:admin or team:admin.'''
+        user, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        org = SeedOrganization.objects.create(title='test org')
+        team = SeedTeam.objects.create(title='test team', organization=org)
+        team.users.add(user)
+
+        data = {
+            'type': 'foo:bar',
+            'object_id': '2',
+            'namespace': 'foo',
+        }
+        permission = SeedPermission.objects.create(**data)
+        team.permissions.add(permission)
+        response = self.client.delete(reverse(
+            'seedteam-permissions-detail', args=[team.id, permission.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_permission_remove_permission_team_admin(self):
+        '''Users with team:admin for that team should be able to remove any
+        permission except for org:admin from that team.'''
+        user, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        org = SeedOrganization.objects.create(title='test org')
+        team = SeedTeam.objects.create(title='test team', organization=org)
+        team2, _ = self.add_permission(user, 'team:admin', team.pk)
+        self.add_permission(user, 'team:admin', team.pk)
+
+        # org:admin
+        permission = SeedPermission.objects.create(
+            type='org:admin', object_id=org.pk, namespace='foo')
+        team.permissions.add(permission)
+        response = self.client.delete(
+            reverse('seedteam-permissions-detail',
+                    args=[team.id, permission.id]))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # other
+        permission = SeedPermission.objects.create(
+            type='foo:bar', object_id='7', namespace='foo')
+        team.permissions.add(permission)
+        response = self.client.delete(
+            reverse('seedteam-permissions-detail',
+                    args=[team.id, permission.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
Resolved in https://github.com/praekelt/seed-auth-api/pull/28/commits/e1b6e70abc42354578779f03b3d08e12db3fce25